clang/lib/CIR/CodeGen/CIRGenFunction.cpp

```diff
@@ -561,7 +558,6 @@ cir::FuncOp CIRGenFunction::generateCode(clang::GlobalDecl gd, cir::FuncOp fn,
       emitImplicitAssignmentOperatorBody(args);
     } else if (body) {
       if (mlir::failed(emitFunctionBody(body))) {
-        fn.erase();
```

Comment:
This change is a bit suspicious, what's happening here?

Reply:
This crashed the openacc-not-implemented test due to a dangling iterator. Is there a point in deleting the function here if codegen failed?

Reply:
Gotcha!  Not sure I remember what a good reason might have been, and I'm fine with removing this. Thanks for giving the extra background, hard to grasp otherwise.